cmake/winml.cmake

```diff
@@ -117,87 +117,57 @@ set_target_properties(winml_lib_telemetry
-target_include_directories(winml_lib_core PRIVATE ${winml_lib_api_dir})                            # needed for generated headers
-target_include_directories(winml_lib_core PRIVATE ${winml_lib_api_core_dir})
-target_include_directories(winml_lib_core PRIVATE ${winml_lib_common_dir}/inc)
-target_include_directories(winml_lib_core PRIVATE ${ONNXRUNTIME_INCLUDE_DIR})
-target_include_directories(winml_lib_core PRIVATE ${ONNXRUNTIME_INCLUDE_DIR}/core/graph)
-target_include_directories(winml_lib_core PRIVATE ${ONNXRUNTIME_ROOT})
-target_include_directories(winml_lib_core PRIVATE ${ONNXRUNTIME_ROOT}/core/graph)
-
-target_include_directories(winml_lib_core PRIVATE ${REPO_ROOT})                                    # needed for dml ep and graphtransformers
-target_include_directories(winml_lib_core PRIVATE ${REPO_ROOT}/cmake/external/eigen)
-target_include_directories(winml_lib_core PRIVATE ${REPO_ROOT}/cmake/external/gsl/include)
-target_include_directories(winml_lib_core PRIVATE ${REPO_ROOT}/cmake/external/onnx)
-target_include_directories(winml_lib_core PRIVATE ${REPO_ROOT}/cmake/external/protobuf/src)
-
-set_target_properties(winml_lib_core
+target_include_directories(winml_adapter PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/winml_api)                   # windows machine learning generated component headers
+target_include_directories(winml_adapter PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/winml_api/comp_generated)    # windows machine learning generated component headers
+target_include_directories(winml_adapter PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/winml/sdk/cppwinrt/include)  # sdk cppwinrt headers
+target_include_directories(winml_adapter PRIVATE ${winml_lib_api_dir})                                    # needed for generated headers
+target_include_directories(winml_adapter PRIVATE ${winml_lib_dir})
+target_include_directories(winml_adapter PRIVATE ${winml_adapter_dir})
+target_include_directories(winml_adapter PRIVATE ${winml_lib_common_dir}/inc)
+
+set_target_properties(winml_adapter
   PROPERTIES
   FOLDER
   ${target_folder})
 
 # Add deps
-add_dependencies(winml_lib_core onnx)
-add_dependencies(winml_lib_core winml_sdk_cppwinrt)
-add_dependencies(winml_lib_core winml_api)
-add_dependencies(winml_lib_core winml_api_native)
-add_dependencies(winml_lib_core winml_api_native_internal)
+add_dependencies(winml_adapter winml_sdk_cppwinrt)
+add_dependencies(winml_adapter winml_api)
+add_dependencies(winml_adapter winml_api_native)
+add_dependencies(winml_adapter winml_api_native_internal)
 
 # Link libraries
-target_link_libraries(winml_lib_core PRIVATE wil)
-target_link_libraries(winml_lib_core PRIVATE onnxruntime_providers_dml)
+target_link_libraries(winml_adapter PRIVATE wil)
+if (onnxruntime_USE_DML)
+  target_link_libraries(winml_adapter PRIVATE ${REPO_PACKAGE_LOCATION}/DirectML.0.0.1/build/DirectML.targets)
+endif(onnxruntime_USE_DML)
+
+# add it to the onnxruntime shared library
+set(onnxruntime_winml windowsapp.lib -WHOLEARCHIVE:$<TARGET_FILE:winml_adapter>)
```

Comment:
>WHOLEARCHIVE [](start = 38, length = 12)

Removing this is on our TODO list

Reply:
yup !

---
In reply to: [354067471](https://github.com/microsoft/onnxruntime/pull/2534#discussion_r354067471) [](ancestors = 354067471)